atmo/clusters/tasks.py

```diff
@@ -94,21 +104,28 @@ def update_clusters_info():
         cluster_mapping[cluster_info['jobflow_id']] = cluster_info
 
     # go through pending clusters and update the state if needed
+    updated_clusters = []
     for cluster in active_clusters:
-        info = cluster_mapping.get(cluster.jobflow_id)
-        # ignore if no info was found for some reason,
-        # the cluster was deleted in AWS but it wasn't deleted here yet
-        if info is None:
-            continue
-
-        # don't update the state if it's equal to the already stored state
-        if info['state'] == cluster.most_recent_status:
-            continue
-
-        # run an UPDATE query for the cluster
-        cluster.most_recent_status = info['state']
-        cluster.save()
-
-        # if not given enqueue a job to update the public IP address
-        if not cluster.master_address:
-            django_rq.enqueue(update_master_address, cluster.id)
+        with transaction.atomic():
+            info = cluster_mapping.get(cluster.jobflow_id)
+            # ignore if no info was found for some reason,
+            # the cluster was deleted in AWS but it wasn't deleted here yet
+            if info is None:
+                continue
+
+            # don't update the state if it's equal to the already stored state
+            if info['state'] == cluster.most_recent_status:
+                continue
+
+            # run an UPDATE query for the cluster
+            cluster.most_recent_status = info['state']
+            cluster.save()
+
+            updated_clusters.append(cluster.identifier)
+
+            # if not given enqueue a job to update the public IP address
+            if not cluster.master_address:
+                transaction.on_commit(
+                    lambda: update_master_address.delay(cluster.id)
+                )
```

Comment:
Would it make sense to only enqueue this if the state is "RUNNING" (or other statuses)? E.g. if the state went from "STARTING" to "BOOTSTRAPPING", then this is a wasted API call b/c the IP address won't be there yet.